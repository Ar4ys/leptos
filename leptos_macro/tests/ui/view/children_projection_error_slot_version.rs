use leptos::*;

#[component]
fn Root(outer: Outer) -> impl IntoView {
    _ = outer;
}

#[slot]
struct Outer {
    children: ChildrenFn,
}

#[component]
fn Inner(children: ChildrenFn) -> impl IntoView {
    _ = children;
}

#[component]
fn Inmost(name: String) -> impl IntoView {
    _ = name;
}

fn main() {
    let name = "Alice".to_string();

    view! {
        <Root>
            <Outer slot>
                <Inner>
                    <Inmost name=name.clone() />
                </Inner>
            </Outer>
        </Root>
    };
}
