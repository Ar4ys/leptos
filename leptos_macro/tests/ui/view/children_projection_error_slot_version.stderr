error[E0525]: expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce`
  --> tests/ui/view/children_projection_error_slot_version.rs:29:17
   |
29 | /                 <Inner>
30 | |                     <Inmost name=name.clone() />
   | |                                  ---- closure is `FnOnce` because it moves the variable `name` out of its environment
31 | |                 </Inner>
   | |                        ^
   | |                        |
   | |________________________this closure implements `FnOnce`, not `Fn`
   |                          the requirement to implement `Fn` derives from here
   |
   = note: required for `Rc<dyn std::ops::Fn() -> Fragment>` to implement `ToChildren<{closure@$DIR/tests/ui/view/children_projection_error_slot_version.rs:29:17: 31:25}>`
